### PR TITLE
fix(ui): don't crash in case of missing schema reference for discriminant definitions

### PR DESCRIPTION
**Pull request by YiuRULE**
### Description

Regression from the `v4.15.0` due of #8213, it may crash in case of missing reference.

(happened on a real world case with an API using `fastapi` and `pydantic`)

### Motivation and Context

It may currently crash for some people who install the latest swagger version

### How Has This Been Tested?
 
Manually testing using the related openapi.json who created the crash in our side.

### My PR contains... 
- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/unit/core/plugins/samples/fn.js [hunk 1]
```diff
@@ -671,6 +671,65 @@ describe("sampleFromSchema", () => {
 
       expect(sampleFromSchema(definition)).toEqual(expected)
     })
+
+    it("shouldn't crash a crash in case of a missing $$ref and should fallback to the default behaviour", () => {
```
Comment (tim-lai) [inline test/unit/core/plugins/samples/fn.js:675]:
```suggestion
    it("should handle case of a missing $$ref by fallback to the default behavior", () => {
```

**Comment by tim-lai** [comment]
@YiuRULE PR merged! Thanks for the contribution, quick fix, and supporting test! released in [v4.15.2](https://github.com/swagger-api/swagger-ui/releases/tag/v4.15.2)
